packages/frontend/src/components/UnderlyingData/UnderlyingDataProvider.tsx

```diff
@@ -0,0 +1,274 @@
+    );
+
+    const viewData = useCallback(
+        (
+            value: ResultRow[0]['value'],
+            meta: TableColumn['meta'],
+            row: ResultRow,
+        ) => {
+            if (meta?.item === undefined) return;
+
+            // We include tables from all fields that appear on the SQL query (aka tables from all columns in results)
+            const rowFields = Object.keys(row);
+            const fieldsInQuery = allFields.filter((field) =>
+                rowFields.includes(getFieldId(field)),
+            );
+            const tablesInQuery = new Set([
+                ...fieldsInQuery.map((field) => field.table),
+                tableName,
+            ]);
+
+            const availableDimensions = allDimensions.filter(
+                (dimension) =>
+                    tablesInQuery.has(dimension.table) &&
+                    !dimension.timeInterval &&
+                    !dimension.hidden,
+            );
+
+            // If we are viewing data from a metric or a table calculation, we filter using all existing dimensions in the table
+            const dimensionFilters =
+                !isField(meta?.item) || isMetric(meta?.item)
+                    ? Object.entries(row).reduce((acc, r) => {
+                          const [
+                              key,
+                              {
+                                  value: { raw },
+                              },
+                          ] = r;
+
+                          const dimensionFilter: FilterRule = {
+                              id: uuidv4(),
+                              target: {
+                                  fieldId: key,
+                              },
+                              operator: FilterOperator.EQUALS,
+                              values: [raw],
+                          };
+                          const isDimension = availableDimensions.find(
+                              (dimension) => getFieldId(dimension) === key,
+                          );
+
+                          if (
+                              isDimension ||
+                              isDateDimension(availableDimensions, key)
+                          ) {
+                              return [...acc, dimensionFilter];
+                          }
+                          return acc;
+                      }, [] as FilterRule[])
+                    : [
+                          {
+                              id: uuidv4(),
+                              target: {
+                                  fieldId: getFieldId(meta?.item),
+                              },
+                              operator: FilterOperator.EQUALS,
+                              values: [value.raw],
+                          },
+                      ];
+
+            const metricFilters = {
+                dimensions: {
+                    ...filters?.dimensions,
+                    id: uuidv4(),
+                    and: dimensionFilters,
```

Comment:
We should merge the existing filters, correct ? 
Related to this, what happens when using `OR` filters ? @TuringLovesDeathMetal 

I assume we need to have the structure:

- AND 
   -  OR
      - filter 1
      - filter 2
   - AND
      - ...filters from underlying data 

[url to pr env](https://lightdash-pr-2904.onrender.com/projects/c2f86a9a-abf1-4d1a-ab6d-62264556847d/tables/orders?create_saved_chart_version=%7B%22tableName%22%3A%22orders%22%2C%22metricQuery%22%3A%7B%22dimensions%22%3A%5B%22orders_is_completed%22%5D%2C%22metrics%22%3A%5B%22orders_average_order_size%22%5D%2C%22filters%22%3A%7B%22dimensions%22%3A%7B%22id%22%3A%2225b85c0c-a215-47be-a566-9e0312f6ae45%22%2C%22and%22%3A%5B%7B%22id%22%3A%22af721828-1f7a-4ca7-b6b0-2d22219f2fc1%22%2C%22target%22%3A%7B%22fieldId%22%3A%22orders_status%22%7D%2C%22operator%22%3A%22equals%22%2C%22values%22%3A%5B%22placed%22%5D%7D%5D%7D%7D%2C%22sorts%22%3A%5B%7B%22fieldId%22%3A%22orders_average_order_size%22%2C%22descending%22%3Atrue%7D%5D%2C%22limit%22%3A500%2C%22tableCalculations%22%3A%5B%5D%2C%22additionalMetrics%22%3A%5B%5D%7D%2C%22pivotConfig%22%3A%7B%22columns%22%3A%5B%22orders_is_completed%22%5D%7D%2C%22tableConfig%22%3A%7B%22columnOrder%22%3A%5B%22orders_is_completed%22%2C%22orders_average_order_size%22%5D%7D%2C%22chartConfig%22%3A%7B%22type%22%3A%22cartesian%22%2C%22config%22%3A%7B%22layout%22%3A%7B%22xField%22%3A%22orders_is_completed%22%2C%22yField%22%3A%5B%22orders_average_order_size%22%5D%7D%2C%22eChartsConfig%22%3A%7B%22series%22%3A%5B%7B%22type%22%3A%22bar%22%2C%22encode%22%3A%7B%22xRef%22%3A%7B%22field%22%3A%22orders_is_completed%22%7D%2C%22yRef%22%3A%7B%22field%22%3A%22orders_average_order_size%22%2C%22pivotValues%22%3A%5B%7B%22field%22%3A%22orders_is_completed%22%2C%22value%22%3Afalse%7D%5D%7D%7D%7D%5D%7D%7D%7D%7D)
![Screenshot 2022-08-15 at 11 13 57](https://user-images.githubusercontent.com/9117144/184618065-6144d376-0696-4d6b-86f4-310aff56ca2d.png)

Reply:
yes! sorry, I didn't even notice this. 

That's the filtering logic I'd expect too. 

So, the filters in the query, then `AND` the filters for the underlying data.

Reply:
Yeah , that was working on a previous version but I think I accidentally removed it on a refactoring... It is working again
![Peek 2022-08-15 13-15](https://user-images.githubusercontent.com/1983672/184625977-93807e87-df38-40b0-bbc1-6c961ae7f732.gif)

Reply:
Niiiiice!

I think in Priyanka's design, there was a little filter tag at the top of the table? Is that still a thing?

Reply:
^let's leave this for now actually! (as in, leave it as-is). 

These smaller things will come up as we use it more :) let's not worry about it now!

Reply:
@rephus @TuringLovesDeathMetal We never added support for nested filter operators. Our query builder applies the same operator to all the rules ( `packages/backend/src/queryBuilder.ts` line 360 ). eg: in this case is using `AND` for the query filters and the underlying data filters causing the underlying query to have 0 results.
 
![Screenshot 2022-08-15 at 13 58 02](https://user-images.githubusercontent.com/9117144/184639776-72d5dc06-5bb2-49c6-a5e8-d113c345b4f3.png)

Reply:
I propose opening a separate ticket to support this scenario.

Reply:
@rephus can you open an issue for this? Or @ZeRego? 

I'm not too sure I understand what's happening

Reply:
https://github.com/lightdash/lightdash/issues/2956